shared/src/main/scala/pl/metastack/metarouter/Router.scala

```diff
@@ -8,42 +8,13 @@ import shapeless._
 import shapeless.ops.hlist._
 import shapeless.poly._
 
-class Router(parsers: List[String => Option[Any]]) {
```

Comment:
Very nice you've gotten rid of this. The solution with pattern matching and unapply is way better: the router is the body of the match statement only, this increases flexibility.